src/background/stores/assessment-card-selection-store.ts

```diff
@@ -0,0 +1,183 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { AssessmentCardSelectionActions } from 'background/actions/assessment-card-selection-actions';
+import { IndexedDBDataKeys } from 'background/IndexedDBDataKeys';
+import { PersistentStore } from 'common/flux/persistent-store';
+import { IndexedDBAPI } from 'common/indexedDB/indexedDB';
+import { Logger } from 'common/logging/logger';
+import { AssessmentCardSelectionStoreData } from 'common/types/store-data/assessment-card-selection-store-data';
+import { forOwn, isEmpty } from 'lodash';
+import { StoreNames } from '../../common/stores/store-names';
+import { RuleExpandCollapseData } from '../../common/types/store-data/card-selection-store-data';
+import {
+    AssessmentCardSelectionPayload,
+    RuleExpandCollapsePayload,
+} from '../actions/action-payloads';
+
+export class AssessmentCardSelectionStore extends PersistentStore<AssessmentCardSelectionStoreData> {
+    constructor(
+        private readonly assessmentCardSelectionActions: AssessmentCardSelectionActions,
+        persistedState: AssessmentCardSelectionStoreData,
+        idbInstance: IndexedDBAPI,
+        logger: Logger,
+        tabId: number,
+        persistStoreData: boolean,
+    ) {
+        super(
+            StoreNames.AssessmentCardSelectionStore,
+            persistedState,
+            idbInstance,
+            IndexedDBDataKeys.assessmentCardSelectionStore(tabId),
+            logger,
+            persistStoreData,
+        );
+    }
+
+    protected addActionListeners(): void {
+        this.assessmentCardSelectionActions.toggleRuleExpandCollapse.addListener(
+            this.toggleRuleExpandCollapse,
+        );
+        this.assessmentCardSelectionActions.toggleCardSelection.addListener(
+            this.toggleCardSelection,
+        );
+        this.assessmentCardSelectionActions.collapseAllRules.addListener(this.collapseAllRules);
+        this.assessmentCardSelectionActions.expandAllRules.addListener(this.expandAllRules);
+        this.assessmentCardSelectionActions.toggleVisualHelper.addListener(this.toggleVisualHelper);
+        this.assessmentCardSelectionActions.getCurrentState.addListener(this.onGetCurrentState);
+        this.assessmentCardSelectionActions.resetFocusedIdentifier.addListener(
+            this.onResetFocusedIdentifier,
+        );
+        this.assessmentCardSelectionActions.navigateToNewCardsView.addListener(
+            this.onNavigateToNewCardsView,
+        );
+    }
+
+    public getDefaultState(): AssessmentCardSelectionStoreData {
+        const defaultValue: AssessmentCardSelectionStoreData = {
+            testKey: {
```

Comment:
I think the testKey state value is being misused throughout the store. With this current implementation there isn't really any point to having a testKey property at all, because we only ever reference `this.state.testKey....`. I think the ideal usage is that the testKey value corresponds to different assessment requirements. So right now, the only testKey in use will be automated checks, but in the future we might have AssessmentCardSelectionStoreData state that looks more like this:

```
{
   'automated-checks': { rules: {....}, visualHelperEnabled: false, focusedResultUid: ''},
   'landmarks': { rules: {....}, visualHelperEnabled: false, focusedResultUid: ''},
   'focus': { rules: {....}, visualHelperEnabled: false, focusedResultUid: ''},
....
}
```

So throughout this store instead of always referencing `this.state.testKey....` it should be more like `this.state['test-key-value-we-get-from-payload']....`. This change will also require us to add the test key to many of the new action payloads that are missing this property right now.